### DISCUSSION TITLE
RDP With NLA on a Azure VM with your Azure AD credentials.

### DISCUSSION THREAD

**Discussion by Cahlingo** [FreeRDP/FreeRDP#8617] [General]
Hello!
I'm currently investigating to RDP from a macOS to a Azure Windows VM. 
I want to RDP to the machine while still having NLA on. Haven't figured out a way to do this.
Does anyone have a clue if FreeRDP supports RDP with NLA on using AAD Credentials. And if so, what's the command?

Im currently trying with 

xfreerdp /u:AzureAD\example@domain.onmicrosoft.com /p:example123 /v:ipadress:host  
tried with two \ aswell. xfreerdp /u:AzureAD\\example@domain.onmicrosoft.com /p:example123 /v:ipadress:host 

If i'm using the RDP file on the mac with my local VM user and pw, it's working obv, but the AAD is not working.

Any solutions?

**Comment by akallabeth** [comment]
* I don´t know if that works if not both systems that are not `AAD joined`?
* You definitely need to use the `dns name` instead of `ip`  to connect (needs to match certificate subject and `kerberos` only works with `dns names` 
* There might be some configuration required (`kerberos` only works well from current `master`, but the `realm` needs to be manually configured for now)

**Comment by fifthdegree** [comment]
Just want to add my two cents in case anyone is interested. The way windows does authentication in these cases is via pku2u using certificates obtained during azure ad registration. So on windows, using the windows sspi module, the "AzureAD\" prefix will tell windows to use these certificates for authentication. Of course, as @akallabeth pointed out, you need to be azure ad joined in order to have those certificates. Now on other platforms, you would first and foremost need to somehow obtain the certificates. Then you would still need a pku2u implementation which we don't currently have. So, no, not really possible from macOS. The other option would be to use Azure AD DS and use standard kerberos
